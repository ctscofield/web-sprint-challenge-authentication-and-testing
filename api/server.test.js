// Write your tests here
const request = require('supertest')
const db = require('../data/dbConfig')
const server = require('./server')

beforeAll(async () => {
  await db.migrate.rollback()
  await db.migrate.latest()
}) //migrate
afterAll(async () => {
  await db.destroy()
})

describe('[GET] /api/jokes', () => {
  test('gets err when hit w/out token', async () => {
    const res = await request(server).get('/api/jokes')
    expect(res.body).toMatchObject({"message": "token required"})
    expect(res.body).toBeTruthy()
  })
  test('snapshot test', async () => {
    const res = await request(server).get('/api/jokes')
    expect(res.body).toMatchSnapshot()
  })
})
