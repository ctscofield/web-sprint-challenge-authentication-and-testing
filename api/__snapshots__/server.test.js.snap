// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[GET] /api/jokes snapshot test 1`] = `
Object {
  "message": "token required",
}
`;
